tests/IResearch/StorageEngineMock.cpp

```diff
@@ -1486,26 +1483,23 @@ size_t TransactionStateMock::abortTransactionCount;
 size_t TransactionStateMock::beginTransactionCount;
 size_t TransactionStateMock::commitTransactionCount;
 
-TransactionStateMock::TransactionStateMock(
-    TRI_vocbase_t& vocbase,
-    arangodb::transaction::Options const& options
-): TransactionState(vocbase, options) {
+// ensure each transaction state has a unique ID
+TransactionStateMock::TransactionStateMock(TRI_vocbase_t& vocbase, arangodb::transaction::Options const& options)
+  : TransactionState(vocbase, 0, options) {
 }
 
 arangodb::Result TransactionStateMock::abortTransaction(arangodb::transaction::Methods* trx) {
   ++abortTransactionCount;
   updateStatus(arangodb::transaction::Status::ABORTED);
   unuseCollections(_nestingLevel);
-  _id = 0; // avoid use of TransactionManagerFeature::manager()->unregisterTransaction(...)
+  // avoid use of TransactionManagerFeature::manager()->unregisterTransaction(...)
 
   return arangodb::Result();
 }
 
 arangodb::Result TransactionStateMock::beginTransaction(arangodb::transaction::Hints hints) {
-  static std::atomic<TRI_voc_tid_t> lastId(0);
 
   ++beginTransactionCount;
-  _id = ++lastId; // ensure each transaction state has a unique ID
```

Comment:
do we no longer use unique transaction ids?

Reply:
in anticipation of cluster wide transactions I now pass in the tid through the constructor.
Using a non-zero ID breaks some iresearch unit-tests